src/main/java/org/manifold/compiler/front/Main.java

```diff
@@ -314,25 +164,78 @@ public Expression visitFunctionInvocationExpression(
   }
 
   @Override
-  public Expression visitTupleValue(ManifoldParser.TupleValueContext context) {
-    // get the expressions resulting from visiting all value entries
-    List<Expression> values = new ArrayList<Expression>();
-    for (ManifoldParser.TupleValueEntryContext subctx
-        : context.tupleValueEntry()) {
-      values.add(visit(subctx.expression()));
+  public Expression visitPrimitiveNodeDefinitionExpression(
```

Comment:
For some reason this is how ANTLR4 generates visitors. I don't really know how to control this code generation.